.github/workflows/check_wifi_fw_patch.yml

```diff
@@ -0,0 +1,41 @@
+name: Verify nRF70 Wi-Fi firmware patch version
+on: pull_request
+env:
+    PATCH_INFO_FILE: 'nrf_wifi/fw_if/umac_if/inc/fw/patch_info.h'
+    PATCH_BIN_BASE: 'nrf_wifi/fw_bins'
+    COMMENT: 'nRF70 Wi-Fi firmware patch version:'
+jobs:
+    verify:
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Extract patch version
+        run: |
+            echo EXP_PATCH_VERSION=${{ env.PATCH_INFO_FILE }} | grep "#define RPU_" | awk '{print $3}' | tr -d '()' | paste -sd "." >> "$GITHUB_OUTPUT"
+        id: patch_version
+      - name: Verify patch version
+        run: |
+          for patch in $(find ${{ env.PATCH_BIN_BASE }} -name 'nrf70.bin'); do
+              echo "Verifying $patch"
+              patch_version=$(hexdump -s 4 -n 4 -e '16/1 "%x " "\n"'  $patch | tr -d ' ' | rev | sed 's/.\{1\}/&./g' | sed 's/.$//')
+              echo "Patch version: $patch_version, expected: ${{ steps.patch_version.outputs.EXP_PATCH_VERSION }}"
+              if [ "${{ steps.patch_version.outputs.EXP_PATCH_VERSION }}" != "$patch_version" ]; then
+                  echo "Patch version mismatch for $patch: $patch_version != ${{ steps.patch_version.outputs.EXP_PATCH_VERSION }}" >> "$COMMENT"
+                  exit 1
+              else
+                  echo "Patch version verified for $patch" >> "$COMMENT"
+              fi
+          done
+      - name: Comment on PR
+        if: ${{ always() }}
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
have you tested this?  

I can't remember if checks in this context have the correct permissions to write comments

Reply:
> Error: Unhandled error: HttpError: Resource not accessible by integration

Token doesn't have permissions, should I use a personal token or should we add `commenting` permissions? @thst-nordic

Reply:
or I have modified to use `pull_request_target` but that requires merging